#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.007979425449207395 0.9914308704298637 0.13038848848032303 0.4548585997615166
  position -1.142239972573004 -0.0887984770165703 0.4748169293907456
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 2 2
}
Robot {
  rotation 0 1 0 0
  children [
    DistanceSensor {
      translation 0 -0.05 0.04
      rotation 0 0 1 -1.5707953071795862
      name "ds1"
      recognitionColors [
        1 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.03 -0.04 0.04
      rotation 0 0 1 -0.9163003061004252
      name "ds2"
      recognitionColors [
        1 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation -0.03 -0.04 0.04
      rotation 0 0 1 -2.2252996938995744
      name "ds3"
      recognitionColors [
        1 0 0
      ]
      type "laser"
    }
    DEF Body Transform {
      translation 0 0 0.04
      children [
        Shape {
          appearance PBRAppearance {
            roughness 1
          }
          geometry Cylinder {
            height 0.05
            radius 0.05
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.05 0 0.03
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
      ]
      endPoint Solid {
        translation 0.04999999967810775 7.998302968757473e-15 0.03002946410012402
        rotation 1.0890263893082967e-10 1 1.0890335785348019e-10 1.5707996938982285
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0.4 0.4 0.4
              roughness 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.05 0 0.03
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
      ]
      endPoint Solid {
        translation -0.04999999966958913 -4.806488272386126e-14 0.03002946440452995
        rotation 1.1577208997980834e-10 1 1.157729154547616e-10 1.5707996938841697
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0.4 0.4 0.4
              roughness 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.025
            }
          }
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics DEF wheel_physics Physics {
        }
      }
    }
  ]
  boundingObject USE Body
  physics Physics {
    mass 1
    centerOfMass [
      0 0 0
    ]
  }
  controller "object_avoider_controller"
}
